src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -431,7 +438,7 @@ private CSharpSyntaxNode ParseExpressionOrPatternForSwitchStatement()
         {
             var savedState = _termState;
             _termState |= TerminatorState.IsExpressionOrPatternInCaseLabelOfSwitchStatement;
-            var pattern = ParsePattern(Precedence.Conditional, whenIsKeyword: true);
+            var pattern = ParsePattern(Precedence.Conditional, inSwitchArmPattern: true);
```

Comment:
one of hte two locations where 'true' is passed in.  Here we're parsing case clauses for switch-statements.